swarm-derive/src/lib.rs

```diff
@@ -125,7 +125,7 @@ fn build_struct(ast: &DeriveInput, data_struct: &DataStruct) -> TokenStream {
                             )
                             .unwrap();
                             let ty = &field.ty;
-                            quote! {#variant(<#ty as NetworkBehaviour>::OutEvent)}
+                            quote! {#variant(<#ty as ::libp2p::swarm::NetworkBehaviour>::OutEvent)}
```

Comment:
```suggestion
                            quote! {#variant(<#ty as #trait_to_impl>::OutEvent)}
```

How about reusing `trait_to_impl`?